velox/dwio/common/Options.h

```diff
@@ -605,10 +606,13 @@ struct WriterOptions {
   std::optional<bool> orcWriterIntegerDictionaryEncodingEnabled{std::nullopt};
   std::optional<bool> orcWriterStringDictionaryEncodingEnabled{std::nullopt};
   std::map<std::string, std::string> serdeParameters;
-  std::optional<uint8_t> parquetWriteTimestampUnit;
   std::optional<uint8_t> zlibCompressionLevel;
   std::optional<uint8_t> zstdCompressionLevel;
 
+  // Writer implementations should provide this function to specify how to
+  // process writer-specific input session and connector configs.
+  virtual void processUserConfigs(const Config*) = 0;
```

Comment:
This needs to be concrete and we need to translate the format agnostic configurations here

Reply:
Why does it need to be concrete?

Reply:
Because it needs to translate format-agnostic options, and the subclasses will handle the format-specific ones